app/src/main/java/net/aiscope/gdd_app/ui/mask/MaskActivity.kt

```diff
@@ -57,38 +58,42 @@ class MaskActivity : AppCompatActivity(), MaskView, CaptureFlow {
 
         presenter.start(diseaseName, imageNameExtra)
 
-        if (savedInstanceState == null) {
-            setBrushDrawableDiseaseStage(brushDiseaseStages[0])
-        }
+        currentDiseaseStage =
+            if (savedInstanceState == null) brushDiseaseStages[0]
+            else savedInstanceState.getParcelable("currentDiseaseStage")!!
+        refreshBrushDrawableColor()
+        photo_mask_view.initBrushDiseaseStage(currentDiseaseStage)
```

Comment:
I'm actually not sure either, sorry. What I can say is that during this implementation I changed the way things were initialised at list a couple of times, trying to make it more understandable/maintainable, as the code was getting too complicated. That probably is a consequence of those changes, but now, looking at it, I guess it would not be needed, the way things are... Do you want us to revert this?

Reply:
The smaller the PR, the better :D So if it's not too big of a trouble, reverting unneeded changes would be great :)

Reply:
Just remembered it actually: photo_mask_view.initBrushDiseaseStage(currentDiseaseStage) didn't exist before - the initialisation of the objects was done in a different, less organised/clear (I hope) way. :-)